### ISSUE TITLE
No response from ImagePicker.launchImageLibraryAsync() 

### ISSUE THREAD

**Issue by henrychao24** [expo/expo#11245]
### Summary of Issue 
No response from ImagePicker.launchImageLibraryAsync()

### Environment
Android bare workflow

### Reproducible Demo
 const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });
    console.log(result);
  };
 
### Steps to Reproduce
1. Create a new Expo project with 'expo init', choose bare js temple project;
2. Install expo-image-picker;
3. Everything works fine if I reload the app. But when I use the back button to exit the app. Then reenter the app with launcher icon, I can still choose image form gallery, but no response sadlly.

**Comment by AdamJNavarro** [comment]
Hey @henrychao24, please provide a runnable example for us to test. In this case, a public repo we can clone and run locally would be ideal. If you can provide that and I can reproduce it on my end, I'll re-open this issue.

Cheers,
Adam

**Comment by henrychao24** [comment]
Hey @AdamJNavarro, I upload a bare example project.
[https://github.com/henrychao24/expo-bare-image-picker-demo.git](url)
Thanks for your help.

**Comment by CoatedMoose** [comment]
I am seeing a similar issue. It seems to be a problem on Android 11 (API level 30).

**Comment by AlbertVilaCalvo** [comment]
I'm experiencing the same problem on Android. The device is a OnePlus 5T, running Android 10. The issue happens sometimes.

I have this code:

```
console.log('pickProductImage')
const pickResult = await ImagePicker.launchImageLibraryAsync({})
console.log('pickProductImage pickResult', pickResult)
```

Sometimes the 2nd log doesn't appear at all. The execution is halted. Nothing is returned.

This has happened right now after going to the Android home screen (by clicking the circle button at the system navigation).

@AdamJNavarro a public repo with the issue was provided by @henrychao24 at https://github.com/henrychao24/expo-bare-image-picker-demo I think you should re-open the issue, because it's real.

**Comment by AlbertVilaCalvo** [comment]
There is a similar issue opened, which was closed too: https://github.com/expo/expo/issues/5934

In particular, see the comment https://github.com/expo/expo/issues/5934#issuecomment-585299535

**Comment by AlbertVilaCalvo** [comment]
Maybe it helps, on my Android Studio logs I see this crashes:

```
2021-07-07 10:21:13.598 28099-28099/? W/LastAccessedStorage: Failed to resume: 
    java.io.FileNotFoundException: name
        at com.android.documentsui.base.DocumentInfo.asFileNotFoundException(DocumentInfo.java:427)
        at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:221)
        at com.android.documentsui.base.DocumentInfo.updateSelf(DocumentInfo.java:205)
        at com.android.documentsui.base.DocumentStack.updateDocuments(DocumentStack.java:240)
        at com.android.documentsui.base.DocumentStack.fromLastAccessedCursor(DocumentStack.java:255)
        at com.android.documentsui.picker.LastAccessedStorage$RuntimeLastAccessedStorage.getLastAccessed(LastAccessedStorage.java:66)
        at com.android.documentsui.picker.ActionHandler.initLoadLastAccessedStack(ActionHandler.java:176)
        at com.android.documentsui.picker.ActionHandler.initLocation(ActionHandler.java:135)
        at com.android.documentsui.picker.PickActivity.onCreate(PickActivity.java:162)
        at android.app.Activity.performCreate(Activity.java:7815)
        at android.app.Activity.performCreate(Activity.java:7804)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1325)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3350)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3514)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2110)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7697)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: java.lang.NullPointerException: name
        at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:133)
        at android.content.ContentResolver.acquireUnstableContentProviderClient(ContentResolver.java:2266)
        at com.android.documentsui.DocumentsApplication.acquireUnstableProviderOrThrow(DocumentsApplication.java:85)
        at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:213)
        at com.android.documentsui.base.DocumentInfo.updateSelf(DocumentInfo.java:205) 
        at com.android.documentsui.base.DocumentStack.updateDocuments(DocumentStack.java:240) 
        at com.android.documentsui.base.DocumentStack.fromLastAccessedCursor(DocumentStack.java:255) 
        at com.android.documentsui.picker.LastAccessedStorage$RuntimeLastAccessedStorage.getLastAccessed(LastAccessedStorage.java:66) 
        at com.android.documentsui.picker.ActionHandler.initLoadLastAccessedStack(ActionHandler.java:176) 
        at com.android.documentsui.picker.ActionHandler.initLocation(ActionHandler.java:135) 
        at com.android.documentsui.picker.PickActivity.onCreate(PickActivity.java:162) 
        at android.app.Activity.performCreate(Activity.java:7815) 
        at android.app.Activity.performCreate(Activity.java:7804) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1325) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3350) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3514) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2110) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7697) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
2021-07-07 10:21:13.598 28099-28099/? E/oid.documentsu: No package ID ff found for ID 0xffffffff.
2021-07-07 10:21:13.605 28099-28099/? W/SelectionTracker: Setting gestureTooltypes is likely to result in unexpected behavior.
2021-07-07 10:21:13.608 28099-28099/? E/oid.documentsu: No package ID ff found for ID 0xffffffff.
2021-07-07 10:21:13.612 28099-28570/? W/RefreshTask: Failed to refresh
    java.lang.NullPointerException: name
        at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:133)
        at android.content.ContentResolver.acquireUnstableContentProviderClient(ContentResolver.java:2266)
        at com.android.documentsui.DocumentsApplication.acquireUnstableProviderOrThrow(DocumentsApplication.java:85)
        at com.android.documentsui.RefreshTask.run(RefreshTask.java:108)
        at com.android.documentsui.RefreshTask.run(RefreshTask.java:43)
        at com.android.documentsui.base.CheckedTask.doInBackground(CheckedTask.java:65)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at com.android.documentsui.ProviderExecutor.run(ProviderExecutor.java:104)
```

**Comment by gprathour** [comment]
I faced the exact same issue in one project. There the solutions was to add the following activity to the AndroidManifest.xml file within <application> tag.

```
<activity
        android:name="com.theartofdev.edmodo.cropper.CropImageActivity"
        android:theme="@style/Base.Theme.AppCompat" />
```

As described [here](https://github.com/expo/expo/tree/master/packages/expo-image-picker).

Also make sure you have required permissions added.

**Comment by CoatedMoose** [comment]
Looks like this got fixed in #13975.

**Comment by mateusjbarbosa** [comment]
Hi,

I am going the same problem on expo-image-picker v12.0.1 with Android 11. The problem only occurs in "production mode". During development `expo start` works correctly.

Logcat error:
`
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: Failed to resume: 
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: java.io.FileNotFoundException: name
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.asFileNotFoundException(DocumentInfo.java:427)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:221)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateSelf(DocumentInfo.java:205)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentStack.updateDocuments(DocumentStack.java:240)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentStack.fromLastAccessedCursor(DocumentStack.java:255)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.LastAccessedStorage$RuntimeLastAccessedStorage.getLastAccessed(LastAccessedStorage.java:66)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.ActionHandler.initLoadLastAccessedStack(ActionHandler.java:176)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.ActionHandler.initLocation(ActionHandler.java:135)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.PickActivity.onCreate(PickActivity.java:162)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Activity.performCreate(Activity.java:8108)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Activity.performCreate(Activity.java:8092)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3481)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3683)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2132)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.os.Looper.loop(Looper.java:250)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.main(ActivityThread.java:7851)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at java.lang.reflect.Method.invoke(Native Method)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:958)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: Caused by: java.lang.NullPointerException: name
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at java.util.Objects.requireNonNull(Objects.java:245)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.content.ContentResolver.acquireUnstableContentProviderClient(ContentResolver.java:2597)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.DocumentsApplication.acquireUnstableProviderOrThrow(DocumentsApplication.java:85)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:213)
03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	... 22 more
`

Thanks.

**Comment by kesylo** [comment]
Also facing the same issue in expo-image-picker 13.3.1 on some android devices. please can someone help fix this once and for all

**Comment by byyoungjin** [comment]
> I faced the exact same issue in one project. There the solutions was to add the following activity to the AndroidManifest.xml file within `<application>` tag.
> 
> ```
> <activity
>         android:name="com.theartofdev.edmodo.cropper.CropImageActivity"
>         android:theme="@style/Base.Theme.AppCompat" />
> ```
> 
> As described [here](https://github.com/expo/expo/tree/master/packages/expo-image-picker).
> 
> Also make sure you have required permissions added.

@kesylo 
This works for me 
maybe documentation needs to be fixed..? 

there is no activity configuration instructed here (which fixed this issue) 
https://github.com/expo/expo/tree/sdk-46/packages/expo-image-picker

It should be like this 
https://github.com/expo/expo/tree/master/packages/expo-image-picker

**Comment by byyoungjin** [comment]
I found this only happens when I hot-reload the app. 

build the app - working 
edit save -> working 
hot reloading -> not working 
re-build ( react-native run-android) -> working 

So I think this would be no problem on production but a little annoying developing the app. 
I should re-build it if I have hot reloaded.

**Comment by emeraldsanto** [comment]
> Hi,
> 
> I am going the same problem on expo-image-picker v12.0.1 with Android 11. The problem occurs in "production mode" only. During development `expo start` works correctly.
> 
> Logcat error:
> 
> ```
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: Failed to resume:
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: java.io.FileNotFoundException: name
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.asFileNotFoundException(DocumentInfo.java:427)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:221)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateSelf(DocumentInfo.java:205)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentStack.updateDocuments(DocumentStack.java:240)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentStack.fromLastAccessedCursor(DocumentStack.java:255)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.LastAccessedStorage$RuntimeLastAccessedStorage.getLastAccessed(LastAccessedStorage.java:66)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.ActionHandler.initLoadLastAccessedStack(ActionHandler.java:176)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.ActionHandler.initLocation(ActionHandler.java:135)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.picker.PickActivity.onCreate(PickActivity.java:162)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Activity.performCreate(Activity.java:8108)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Activity.performCreate(Activity.java:8092)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3481)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3683)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2132)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.os.Looper.loop(Looper.java:250)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.app.ActivityThread.main(ActivityThread.java:7851)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at java.lang.reflect.Method.invoke(Native Method)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:958)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: Caused by: java.lang.NullPointerException: name
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at java.util.Objects.requireNonNull(Objects.java:245)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at android.content.ContentResolver.acquireUnstableContentProviderClient(ContentResolver.java:2597)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.DocumentsApplication.acquireUnstableProviderOrThrow(DocumentsApplication.java:85)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	at com.android.documentsui.base.DocumentInfo.updateFromUri(DocumentInfo.java:213)
> 03-02 23:00:07.331 25795 25795 W LastAccessedStorage: 	... 22 more
> ```
> 
> Thanks.

I get the exact same stack trace on SDK 46 with `expo-image-picker@13.3.1` except when I disable ProGuard, it must be related to some classes being obfuscated by mistake. FWIW I tried using the rules described [here](https://github.com/expo/expo/issues/19509#issuecomment-1278223157) but no dice :(

**Comment by olstamar** [comment]
I'm running into a similar issue. Using `expo-image-picker@13.3.1`. Works fine on iOS but I'm running into issues on android where the users cannot select image from gallery. Tried disabling ProGuard as mentioned above. I set `"enableProguardInReleaseBuilds": false` and then I see no issue with selecting image on an android device.

So it can seem as the above comments suggests that ProGuard obfuscates the code in a way that makes it difficult to select image file.

It seems it is possible to add some ProGuards rules with `extraProguardRules` param but I'm not sure how to use this one at the moment.
